src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -957,7 +957,28 @@ protected bool QualifiedIdentifier(out int identifierLength)
 
         protected bool NamespaceOrTypeName()
         {
-            if (Optional(CSharpSymbolType.Identifier) || Optional(CSharpSymbolType.Keyword))
+            if (Optional(CSharpSymbolType.LeftParenthesis))
+            {
+                while (!Optional(CSharpSymbolType.RightParenthesis) && !EndOfFile)
+                {
+                    Optional(CSharpSymbolType.WhiteSpace);
+
+                    if (!NamespaceOrTypeName())
+                    {
+                        return false;
+                    }
+
+                    Optional(CSharpSymbolType.WhiteSpace);
+                    Optional(CSharpSymbolType.Identifier);
+                    Optional(CSharpSymbolType.WhiteSpace);
+                    Optional(CSharpSymbolType.Comma);
+                }
+
```

Comment:
Oops, missed this. Can have optional whitespace before the questionmark too. That's a little more tricky though. You only want to consume the whitespace if the next symbol is a question mark. Should do the same thing with our other `nullable` support below.